validate/validate.go

```diff
@@ -298,6 +302,17 @@ func verifyBody(fn *wasm.FunctionSig, body *wasm.FunctionBody, module *wasm.Modu
 			if err != nil {
 				return vm, err
 			}
+			tableIndex, err := vm.fetchByte()
+			if err != nil {
+				return vm, err
+			}
+			if tableIndex != 0x00 {
+				return vm, errors.New("validate: table index in call_indirect must be 0")
+			}
+
+			if index >= uint32(len(module.Types.Entries)) {
```

Comment:
perhaps move this check closer to the type index value extraction? (ie: to line 305)

Reply:
I casually added this, since I found `index` used in line 317 is not bound checked.  so it is reasonable to place this check here just before the place of use?

Reply:
that stands to reason as well.